File: src/main/org/nlogo/prim/etc/_linklength.java
package org.nlogo.prim.etc;

import org.nlogo.agent.Link;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.api.Syntax;

public final strictfp class _linklength extends Reporter {
  @Override
  public Syntax syntax() {
    return Syntax.reporterSyntax(Syntax.NumberType(), "---L");
  }

  @Override
  public Object report(Context context) {
    return report_1(context);
  }

  public double report_1(Context context) {
    return ((Link) context.agent).size();
  }
}

File: src/main/prim/etc/_linklength.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.agent.Link
import org.nlogo.core.{ Syntax, SyntaxJ }
import org.nlogo.nvm.{ Context, Reporter }

class _linklength extends Reporter {
  override def syntax =
    SyntaxJ.reporterSyntax(Syntax.NumberType, "---L")
  override def report(context: Context): java.lang.Double =
    Double.box(report_1(context))
  def report_1(context: Context): Double =
    context.agent.asInstanceOf[Link].size
}
